src/game/RandomizationOfShipsPositions.js

```diff
@@ -86,40 +88,53 @@ const positionShipHorizontally = (shipIndex, direction) => {
   if (
     gameBoardArray[position[0]]
       .slice(position[1], position[1] + SHIPS_LIST[shipIndex].size)
-      .some((el) => el.isEmpty === false)
+      .some((el) => {
+        return typeof el === Object && el.isEmpty === false;
+      })
   ) {
     positionShipHorizontally(shipIndex);
   } else {
-    gameBoardArray[position[0]].fill(
-      {
+    for (let i = 0; i < SHIPS_LIST[shipIndex].size; i++) {
+      gameBoardArray[position[0]][position[1] + i] = {
         isEmpty: false,
         shipName: SHIPS_LIST[shipIndex].shipName,
-      },
-      position[1],
-      position[1] + SHIPS_LIST[shipIndex].size
-    );
+        fieldName: fieldsNamesArray[position[0] * 10 + i],
+        hit: false,
+      };
+    }
   }
 };
 
 export const addShipsToBoard = (isPlayer) => {
   createGameBoardArray();
-  placeShipsOnBoard();
-  GAME_BOARD[isPlayer ? 0 : 1].innerHTML = "";
-  let gameBoardForMap = gameBoardArray.flat(Infinity);
-  gameBoardForMap.map((_, i) => {
-    let newField = document.createElement("div");
-    newField.classList.add(`board__field`);
-    if (gameBoardForMap[i].isEmpty === false) {
-      newField.classList.add(`ship`);
-    }
-    if (gameBoardForMap[i].shipName != "") {
-      newField.setAttribute(
-        "data-ship-name-board",
-        gameBoardForMap[i].shipName
-      );
-    }
-    newField.setAttribute("data-field-name", fieldsNamesArray[i]);
-    newField.dataset.id = i;
-    GAME_BOARD[isPlayer ? 0 : 1].appendChild(newField);
-  });
+  const gameBoard = placeShipsOnBoard();
+  if (isPlayer) {
+    GAME_BOARD[isPlayer ? 0 : 1].innerHTML = "";
+    let gameBoardForMap = gameBoardArray.flat(Infinity);
+    gameBoardForMap.map((_, i) => {
+      let newField = document.createElement("div");
+      newField.classList.add(`board__field`);
+      if (gameBoardForMap[i].isEmpty === false) {
+        newField.classList.add(`ship`);
+      }
+      if (gameBoardForMap[i].shipName != "") {
+        newField.setAttribute(
+          "data-ship-name-board",
+          gameBoardForMap[i].shipName
+        );
+      }
+      newField.setAttribute("data-field-name", fieldsNamesArray[i]);
+      newField.dataset.id = i;
+      GAME_BOARD[isPlayer ? 0 : 1].appendChild(newField);
+    });
+  }
+  return separateShipsFromEmptyFields(gameBoard);
+};
+
+const separateShipsFromEmptyFields = (board) => {
+  let emptyFields = board
+    .flat(Infinity)
+    .filter((field) => field.isEmpty === true);
```

Comment:
Isn't `field.isEmpty` enough? If it is true it will return `true` if it will be `false` it will return false :)

Reply:
it's enough 🤦

Reply:
fixed